### PR TITLE
Add a simple roundtrip fuzz target

### PR DESCRIPTION
**Pull request by bschwind**
This isn't really a complete addition, more of a jumpstart for anyone who wants to finish adding code fuzzing to this repo. It currently panics pretty early, I'm not sure if that's me not using the library correctly or the fuzzer catching legitimately failing cases.

More references can be found from [this comment](https://www.reddit.com/r/rust/comments/ge3kee/lzfear_a_purerust_nounsafe_lz4_implementation/fpl8sp7/)

### CODE REVIEW

**Comment by main--** [comment]
Thank you! It was a really simple bug (12279fc1acf5078cec443d65bb22a6f2a9eb5229) that I introduced when refactoring the header parsing code and I hadn't tested any roundtripping since (the bug made it impossible to decompress anything at all).

**Comment by bschwind** [comment]
Great! Thanks for finishing off the implementation :)

File: fuzz/fuzz_targets/roundtrip_fuzz.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        .compress(input, &mut output) //_with_size(input, &mut output)
+        .expect("Could not compress input data");
+
+    let mut lz4_reader = LZ4FrameReader::new(Cursor::new(output))
+        .expect("Could not create frame reader")
+        .into_read();
+
+    let mut roundtripped = Vec::new();
+    lz4_reader.read_to_end(&mut roundtripped).expect("Could not read decompressed data");
+    assert!(roundtripped.iter().eq(data));
```
Comment (Shnatsel) [inline fuzz/fuzz_targets/roundtrip_fuzz.rs:22]: It's probably slightly faster to compare slices than iterators: `assert!(roundtripped.as_slice() == data);` or `assert!(&roundtripped == data);`

**Comment by Shnatsel** [comment]
Running this for 50 million executions didn't reveal any issues. Great!

**Pull request merged**